contracts/BodhiCandy.sol

```diff
@@ -0,0 +1,65 @@
+pragma solidity ^0.4.18;
+
+import './libs/SafeMath.sol';
+
+contract BodhiCandy {
+    using SafeMath for uint256;
+
+    // Amount in Satoshi to deposit
+    uint256 public constant depositAmount = 1000000;
+
+    // Amount of blocks it takes for the last person to deposit to win if no others deposit within the length.
+    uint256 public constant winningBlockLength = 30;
+
+    address public lastDepositer;
+    uint256 public lastDepositBlock;
+    uint256 public currentBalance;
+
+    event UserWon(address indexed winner, uint256 amountWon);
+    event Returned(uint256 returned);
+
+    function deposit() external payable {
+        require(msg.value >= depositAmount);
+
+        // Calculate any amount over the depositAmount to return to user
+        uint256 excessDeposit = 0;
+        if (msg.value > depositAmount) {
+            excessDeposit = msg.value.sub(depositAmount);
+        }
+
+        // Last depositer wins
+        if (lastDepositer != address(0) 
+            && lastDepositBlock != 0 
+            && block.number.sub(lastDepositBlock) >= winningBlockLength) {
+
+            uint256 amountWon = currentBalance.add(excessDeposit);
+            currentBalance = 0;
+
+            if (amountWon > 0) {
+                lastDepositer.transfer(amountWon);
+                UserWon(lastDepositer, amountWon);
+
+                lastDepositer = msg.sender;
+                lastDepositBlock = block.number;
+                currentBalance = depositAmount;
```

Comment:
The current three lines can be moved out of the `if-else` checking

Reply:
It should stay in the if block to prevent reentrancy attack. I am setting `currentBalance` to 0 to prevent reentrancy, but if I set `currentBalance = depositAmount` outside of the if-else block, it will create a reentrancy hole again.

So in regards to my comment above my if checking statements will be this (making the change now):
```
if (lastDepositer != address(0) 
    && lastDepositBlock != 0 
    && block.number.sub(lastDepositBlock) >= winningBlockLength
    && currentBalance > 0) {
```

example of reentrancy attack with your suggested change:
- user last bet 30 blocks ago, and there is enough qtum to win (ie. 0.5 qtum) from the contract
- user knows they can call the contract to win now, but attacks it with reentrancy
- 1st run through contract, hits first if block > `currentBalance = 0` > transfer() executed > after if-else block, `currentBalance = depositAmount` with the change > so `currentBalance` now equals `depositAmount`
- 2nd run through contract (reentrant attack) > `currentBalance > 0` so it will go to the first block again of the if-else > amountWon will be > 0 so the same user can withdraw again (and again and again)